test-cloud/test-cloud-performance/src/test/java/org/kie/cloud/integrationtests/s2i/KieServerS2iJbpmEJBTimerPerfIntegrationTest.java

```diff
@@ -0,0 +1,358 @@
+                .withKieServer(kieServerS2Isettings)
+                .build();
+    }
+
+    @Before
+    public void setUp() {
+        
+        if (SCALE_COUNT > 1) {
+            logger.info("starting to scale");
+            scaleKieServerTo(SCALE_COUNT);
+            logger.info("scaled to {} pods", SCALE_COUNT);
+        }
+        
+        deploymentScenario.getKieServerDeployments().get(0).setRouterTimeout(Duration.ofMinutes(ROUTER_TIMEOUT));
+        deploymentScenario.getKieServerDeployments().get(0).setRouterBalance(ROUTER_BALANCE);
+        
+        kieServicesClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployments().get(0), Duration.ofMinutes(60).toMillis());
+        
+        logger.info("Setting timeout for kieServerClient to {}", Duration.ofMinutes(60).toMillis());
+        
+        processServicesClient = KieServerClientProvider.getProcessClient(deploymentScenario.getKieServerDeployments().get(0));
+        queryServicesClient = KieServerClientProvider.getQueryClient(deploymentScenario.getKieServerDeployments().get(0));
+    }
+
+    @After
+    public void deleteRepo() {
+        Git.getProvider().deleteGitRepository(repositoryName);
+        deploymentScenario.getKieServerDeployments().get(0).resetRouterTimeout();        
+    }
+
+    @Test
+    @Category(Performance.class)
+    public void testContainerAfterExecServerS2IStart() throws IOException {
+        List<KieContainerResource> containers = kieServicesClient.listContainers().getResult().getContainers();
+        assertThat(containers).isNotNull().hasSize(1);
+
+        KieContainerResource container = containers.get(0);
+        assertThat(container).isNotNull();
+        assertThat(container.getContainerId()).isNotNull().isEqualTo(CONTAINER_ID);
+
+        ReleaseId containerReleaseId = container.getResolvedReleaseId();
+        assertThat(containerReleaseId).isNotNull();
+        assertThat(containerReleaseId.getGroupId()).isNotNull().isEqualTo(PROJECT_GROUP_ID);
+        assertThat(containerReleaseId.getArtifactId()).isNotNull().isEqualTo(DEFINITION_PROJECT_NAME);
+        assertThat(containerReleaseId.getVersion()).isNotNull().isEqualTo(DEFINITION_PROJECT_VERSION);
+
+        logger.info("============================= STARTING SCENARIO =============================");
+        runSingleScenario();
+        logger.info("============================= SCENARIO COMPLETE =============================");
+    
+        logger.info("============================= GATHERING STATISTICS =============================");
+        gatherAndAssertStatistics();
+        writeCSV();
+        logger.info("============================= STATISTICS GATHERED =============================");       
+    }
+
+    private void writeCSV() throws IOException {
+        List<String> headerCompleted = new ArrayList<String>();
+        for (int i=1; i<=pods.size(); i++) {
+            headerCompleted.add("Completed Pod"+i);
+        }
```

Comment:
Is this used anywhere? It was probably meant to be used as a header?

Reply:
No, removed as this was for an attempt to add headers (but the Apache CSV library doesn't allow to have a variable number of headers, as this case)

Reply:
Well, I think that this is only about the number of columns and this is for one run of the tests fixed, so it shouldn't be an issue to add headers, should it? But we can improve that in a separate PR if it is for a longer debate.

Reply:
Yes, we can leave it as a future enhancement. I was trying (without success) to add a variable number of headers but it seems like [Apache library](http://commons.apache.org/proper/commons-csv/user-guide.html#Printing_with_headers) is not prepared to.